examples/xds-management-server/pkg/snapshot/snapshot.go

```diff
@@ -0,0 +1,119 @@
+package snapshot
+
+import (
+	"context"
+	"time"
+
+	"quilkin.dev/xds-management-server/pkg/cluster"
+	"quilkin.dev/xds-management-server/pkg/filterchain"
+
+	"github.com/envoyproxy/go-control-plane/pkg/cache/v3"
+	log "github.com/sirupsen/logrus"
+
+	"quilkin.dev/xds-management-server/pkg/resources"
+)
+
+// RunSnapshotUpdater starts a goroutine that listens for resource updates,
+// uses the updates to generate an xds config snapshot and updates the snapshot from the provided channel and generates xds config snapshots.
+// cache with the latest snapshot for each connected node.
+// It returns the snapshot cache which can be passed on to the xds server impl.
+func RunSnapshotUpdater(
```

Comment:
Looks like everywhere else (still digging in, but I think this is true), you've used the pattern of New() to initalise and Run() to start the go routine (I really like this pattern, which you have likely noticed 😁)

For the sake of consistency, I think it would be nice to stick to that here too.

Also, if you choose to do the time mock, it's easier to overwrite the local clock variable at test time without having to change the constructor.